happi/audit.py

```diff
@@ -0,0 +1,424 @@
+"""
+This module defines the ``happi audit`` command line utility
+
+If the --file option is not provided, then it will take the path from
+Happi configuration file using the Client.find_config method,
+otherwise it will use the path specified after --file
+"""
+from abc import ABCMeta, abstractmethod
+from configparser import ConfigParser
+import logging
+import os
+import sys
+import subprocess
+import re
+from happi.client import Client
+from happi.loader import import_class, fill_template
+from happi.containers import registry
+from enum import Enum
+
+logger = logging.getLogger(__name__)
+
+
+class ReportCode(Enum):
+    """
+    Class to define report codes
+    """
+    SUCCESS = 0
+    INVALID = 1
+    MISSING = 2
+    EXTRAS = 3
+    NO_CODE = 9
+
+
+class Command(object, metaclass=ABCMeta):
```

Comment:
```suggestion
class Command(abc.ABC):
```

Metaclasses are confusing - the less the better: https://docs.python.org/3.8/library/abc.html#abc.ABC

Separately, I'm clearly biased but I've had good luck with the following pattern: https://github.com/slaclab/pytmc/blob/master/pytmc/bin/pytmc.py#L27

Have a top-level command, let the sub-commands build their parsers and return them to you, associate a function and pass the parsed args in as keywords. 🤷‍♂️ 

I do like that you get a registry of subcommands for free with this `Command` class, though.